cmake build on github needs to build fortran too

Seems like the fortran build on the cmake github workflow is not being built, but it needs to be.

@jedwards4b the cmake build still required netcdf-fortran:

```
+ cmake -Wno-dev -DNetCDF_C_LIBRARY=/usr/lib/x86_64-linux-gnu/libnetcdf.so -DNetCDF_C_INCLUDE_DIR=/usr/include -DPnetCDF_PATH=/home/runner/pnetcdf -DPIO_ENABLE_FORTRAN=On -DPIO_HDF5_LOGGING=On -DPIO_USE_MALLOC=On -DPIO_ENABLE_LOGGING=On -DPIO_ENABLE_TIMING=Off ..
19
-- The C compiler identification is GNU 7.5.0
20
-- The Fortran compiler identification is GNU 7.5.0
21
-- Detecting C compiler ABI info
22
-- Detecting C compiler ABI info - done
23
-- Check for working C compiler: /usr/bin/mpicc - skipped
24
-- Detecting C compile features
25
-- Detecting C compile features - done
26
-- Detecting Fortran compiler ABI info
27
-- Detecting Fortran compiler ABI info - done
28
-- Check for working Fortran compiler: /usr/bin/mpifort - skipped
29
-- Checking whether /usr/bin/mpifort supports Fortran 90
30
-- Checking whether /usr/bin/mpifort supports Fortran 90 - yes
31
-- Importing CMake_Fortran_utils
32
-- Importing CMake_Fortran_utils - success
33
-- Found MPI_C: /usr/bin/mpicc (found version "3.1") 
34
-- Found MPI_Fortran: /usr/bin/mpifort (found version "3.1") 
35
-- Found MPI: TRUE (found version "3.1")  
36
-- Looking for sys/types.h
37
-- Looking for sys/types.h - found
38
-- Looking for stdint.h
39
-- Looking for stdint.h - found
40
-- Looking for stddef.h
41
-- Looking for stddef.h - found
42
-- Check size of MPI_Offset
43
-- Check size of MPI_Offset - done
44
-- Found NetCDF_C: /usr/lib/x86_64-linux-gnu/libnetcdf.so  
45
-- Checking NetCDF version
46
-- Checking NetCDF version - 4.6.0./*!<
47
-- Checking whether NetCDF has parallel support
48
-- Checking whether NetCDF has parallel support - no
49
-- Looking for nc_set_log_level
50
Fortran Library build is On
51
-- Looking for nc_set_log_level - not found
52
CMake Error at /usr/local/share/cmake-3.19/Modules/FindPackageHandleStandardArgs.cmake:218 (message):
53
-- Configuring incomplete, errors occurred!
54
  Could NOT find NetCDF_Fortran (missing: NetCDF_Fortran_LIBRARY
55
See also "/home/runner/work/ParallelIO/ParallelIO/build/CMakeFiles/CMakeOutput.log".
56
  NetCDF_Fortran_INCLUDE_DIR)
57
See also "/home/runner/work/ParallelIO/ParallelIO/build/CMakeFiles/CMakeError.log".
58
Call Stack (most recent call first):
59
  /usr/local/share/cmake-3.19/Modules/FindPackageHandleStandardArgs.cmake:577 (_FPHSA_FAILURE_MESSAGE)
60
  cmake/LibFind.cmake:306 (find_package_handle_standard_args)
61
  cmake/FindNetCDF.cmake:49 (find_package_component)
62
  CMakeLists.txt:245 (find_package)
```

--- a/.github/workflows/cmake_netcdf-4.7.4_hdf5-1.10.7_pnetcdf-12.1_ncint_mpich-3.3_asan.yml
+++ b/.github/workflows/cmake_netcdf-4.7.4_hdf5-1.10.7_pnetcdf-12.1_ncint_mpich-3.3_asan.yml
@@ -140,6 +140,7 @@ jobs:
         source .bashrc
         export CC=/home/runner/mpich/bin/mpicc
         export FC=/home/runner/mpich/bin/mpifort
+        export LD_LIBRARY_PATH=/home/runner/netcdf/lib:/home/runner/pnetcdf/lib:$LD_LIBRARY_PATH
         rm -rf b1
         mkdir b1
         cd b1

--- a/.github/workflows/cmake_netcdf-4.7.4_hdf5-1.10.7_pnetcdf-12.1_ncint_mpich-3.3_asan.yml
+++ b/.github/workflows/cmake_netcdf-4.7.4_hdf5-1.10.7_pnetcdf-12.1_ncint_mpich-3.3_asan.yml
@@ -144,7 +144,7 @@ jobs:
         rm -rf b1
         mkdir b1
         cd b1
-        cmake -Wno-dev -DNetCDF_C_LIBRARY=/home/runner/netcdf-c/lib/libnetcdf.so -DNetCDF_C_INCLUDE_DIR=/home/runner/netcdf-c/include -DPnetCDF_PATH='/home/runner/pnetcdf' -DPIO_ENABLE_FORTRAN=Off -DPIO_HDF5_LOGGING=On -DPIO_USE_MALLOC=On -DPIO_ENABLE_LOGGING=On -DPIO_ENABLE_TIMING=Off ..
+        cmake -Wno-dev -DNetCDF_C_LIBRARY=/home/runner/netcdf-c/lib/libnetcdf.so -DNetCDF_C_INCLUDE_DIR=/home/runner/netcdf-c/include -DPnetCDF_PATH=/home/runner/pnetcdf -DPIO_ENABLE_FORTRAN=On -DPIO_ENABLE_LOGGING=On -DPIO_ENABLE_TIMING=Off -DNetCDF_Fortran_LIBRARY=/home/runner/netcdf-fortran/lib/libnetcdff.so -DNetCDF_Fortran_INCLUDE_DIR=/home/runner/netcdf-fortran/include ..
         make VERBOSE=1
         make tests VERBOSE=1
         ctest -VV

--- a/.github/workflows/cmake_netcdf-4.7.4_hdf5-1.10.7_pnetcdf-12.1_ncint_mpich-3.3_asan.yml
+++ b/.github/workflows/cmake_netcdf-4.7.4_hdf5-1.10.7_pnetcdf-12.1_ncint_mpich-3.3_asan.yml
@@ -147,4 +147,4 @@ jobs:
         cmake -Wno-dev -DNetCDF_C_LIBRARY=/home/runner/netcdf-c/lib/libnetcdf.so -DNetCDF_C_INCLUDE_DIR=/home/runner/netcdf-c/include -DPnetCDF_PATH=/home/runner/pnetcdf -DPIO_ENABLE_FORTRAN=On -DPIO_ENABLE_LOGGING=On -DPIO_ENABLE_TIMING=Off -DNetCDF_Fortran_LIBRARY=/home/runner/netcdf-fortran/lib/libnetcdff.so -DNetCDF_Fortran_INCLUDE_DIR=/home/runner/netcdf-fortran/include ..
         make VERBOSE=1
         make tests VERBOSE=1
-        ctest -VV
+        ctest -VV -I 22,22

--- a/.github/workflows/cmake_netcdf-4.7.4_hdf5-1.10.7_pnetcdf-12.1_ncint_mpich-3.3_asan.yml
+++ b/.github/workflows/cmake_netcdf-4.7.4_hdf5-1.10.7_pnetcdf-12.1_ncint_mpich-3.3_asan.yml
@@ -140,14 +140,11 @@ jobs:
         source .bashrc
         export CC=/home/runner/mpich/bin/mpicc
         export FC=/home/runner/mpich/bin/mpifort
-        export LD_LIBRARY_PATH=/home/runner/netcdf/lib:/home/runner/pnetcdf/lib:$LD_LIBRARY_PATH
+        export LD_LIBRARY_PATH=/home/runner/netcdf-c/lib:/home/runner/pnetcdf/lib:$LD_LIBRARY_PATH
         rm -rf b1
         mkdir b1
         cd b1
-        cmake -Wno-dev -DNetCDF_C_LIBRARY=/home/runner/netcdf/lib/libnetcdf.so -DNetCDF_C_INCLUDE_DIR=/usr/include -DPnetCDF_PATH='/home/runner/pnetcdf' -DPIO_ENABLE_FORTRAN=Off -DPIO_HDF5_LOGGING=On -DPIO_USE_MALLOC=On -DPIO_ENABLE_LOGGING=On -DPIO_ENABLE_TIMING=Off ..
+        cmake -Wno-dev -DNetCDF_C_LIBRARY=/home/runner/netcdf-c/lib/libnetcdf.so -DNetCDF_C_INCLUDE_DIR=/home/runner/netcdf-c/include -DPnetCDF_PATH='/home/runner/pnetcdf' -DPIO_ENABLE_FORTRAN=Off -DPIO_HDF5_LOGGING=On -DPIO_USE_MALLOC=On -DPIO_ENABLE_LOGGING=On -DPIO_ENABLE_TIMING=Off ..
         make VERBOSE=1
         make tests VERBOSE=1
         ctest -VV
-        
-        
-        

--- a/.github/workflows/cmake_netcdf-4.7.4_hdf5-1.10.7_pnetcdf-12.1_ncint_mpich-3.3_asan.yml
+++ b/.github/workflows/cmake_netcdf-4.7.4_hdf5-1.10.7_pnetcdf-12.1_ncint_mpich-3.3_asan.yml
@@ -19,7 +19,7 @@ jobs:
     - uses: actions/checkout@v2
     - name: Installs
       run: |
-        sudo apt-get install doxygen graphviz wget gfortran libjpeg-dev libz-dev cmake
+        sudo apt-get install doxygen graphviz wget gfortran libjpeg-dev libz-dev cmake gfortran
     - name: cache-mpich
       id: cache-mpich
       uses: actions/cache@v2

--- a/.github/workflows/cmake_netcdf-4.7.4_hdf5-1.10.7_pnetcdf-12.1_ncint_mpich-3.3_asan.yml
+++ b/.github/workflows/cmake_netcdf-4.7.4_hdf5-1.10.7_pnetcdf-12.1_ncint_mpich-3.3_asan.yml
@@ -1,0 +1,152 @@
+name: cmake_netcdf-4.7.4_hdf5-1.10.7_pnetcdf-12.1_ncint_mpich-3.3
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    env:
+      CPPFLAGS: "-I/home/runner/mpich/include -I/home/runner/hdf5/include -I/home/runner/netcdf-c/include -I/home/runner/netcdf-fortran/include -I/home/runner/pnetcdf/include"
+      LDFLAGS: "-L/home/runner/mpich/lib -L/home/runner/hdf5/lib -L/home/runner/netcdf-c/lib -L/home/runner/netcdf-fortran/lib -L/home/runner/pnetcdf/lib"
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Installs
+      run: |
+        sudo apt-get install doxygen graphviz wget gfortran libjpeg-dev libz-dev cmake
+    - name: cache-mpich
+      id: cache-mpich
+      uses: actions/cache@v2
+      with:
+        path: ~/mpich
+        key: mpich-${{ runner.os }}-3.3.2
+
+    - name: build-mpich
+      if: steps.cache-mpich.outputs.cache-hit != 'true'
+      run: |
+        wget http://www.mpich.org/static/downloads/3.3.2/mpich-3.3.2.tar.gz &> /dev/null
+        tar -xzf mpich-3.3.2.tar.gz
+        pushd mpich-3.3.2
+        ./configure --prefix=/home/runner/mpich
+        make
+        sudo make install
+        popd
+    - name: cache-hdf5
+      id: cache-hdf5
+      uses: actions/cache@v2
+      with:
+        path: ~/hdf5
+        key: hdf5-${{ runner.os }}-1.10.7-mpich-3.3.2
+
+    - name: build-hdf5
+      if: steps.cache-hdf5.outputs.cache-hit != 'true'
+      run: |
+        set -x
+        export PATH="/home/runner/mpich/bin:$PATH"
+        export CC=/home/runner/mpich/bin/mpicc
+        wget https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.10/hdf5-1.10.7/src/hdf5-1.10.7.tar.gz &> /dev/null
+        tar -xzf hdf5-1.10.7.tar.gz
+        pushd hdf5-1.10.7
+        ./configure --prefix=/home/runner/hdf5 --enable-parallel --disable-tools --disable-fortran --disable-cxx --enable-parallel-tests
+        make
+        sudo make install
+        popd
+    - name: cache-netcdf-c
+      id: cache-netcdf-c
+      uses: actions/cache@v2
+      with:
+        path: ~/netcdf-c
+        key: netcdf-c-${{ runner.os }}-4.7.4-mpich-3.3.2-2
+
+    - name: build-netcdf-c
+      if: steps.cache-netcdf-c.outputs.cache-hit != 'true'
+      run: |
+        set -x
+        export PATH="/home/runner/mpich/bin:$PATH"
+        export CC=/home/runner/mpich/bin/mpicc
+        wget https://www.unidata.ucar.edu/downloads/netcdf/ftp/netcdf-c-4.7.4.tar.gz &> /dev/null
+        tar -xzf netcdf-c-4.7.4.tar.gz
+        pushd netcdf-c-4.7.4
+        ./configure --prefix=/home/runner/netcdf-c --disable-dap --disable-utilities
+        make -j
+        sudo make install
+        popd
+    - name: cache-netcdf-fortran
+      id: cache-netcdf-fortran
+      uses: actions/cache@v2
+      with:
+        path: ~/netcdf-fortran
+        key: netcdf-fortran-${{ runner.os }}-4.5.3-mpich-3.3.2-2
+
+    - name: build-netcdf-fortran
+      if: steps.cache-netcdf-fortran.outputs.cache-hit != 'true'
+      run: |
+        set -x
+        export PATH="/home/runner/mpich/bin:$PATH"
+        export CC=/home/runner/mpich/bin/mpicc
+        wget https://github.com/Unidata/netcdf-fortran/archive/v4.5.3.tar.gz &> /dev/null
+        tar -xzf v4.5.3.tar.gz
+        pushd netcdf-fortran-4.5.3
+        ./configure --prefix=/home/runner/netcdf-fortran
+        make -j
+        sudo make install
+        popd
+    - name: cache-pnetcdf
+      id: cache-pnetcdf
+      uses: actions/cache@v2
+      with:
+        path: ~/pnetcdf
+        key: pnetcdf-${{ runner.os }}-1.12.1-mpich-3.3.2
+
+    - name: build-pnetcdf
+      if: steps.cache-pnetcdf.outputs.cache-hit != 'true'
+      run: |
+        set -x
+        export PATH="/home/runner/mpich/bin:$PATH"
+        export CC=/home/runner/mpich/bin/mpicc
+        wget https://parallel-netcdf.github.io/Release/pnetcdf-1.12.1.tar.gz  &> /dev/null
+        tar -xzf pnetcdf-1.12.1.tar.gz
+        pushd pnetcdf-1.12.1
+        ./configure --prefix=/home/runner/pnetcdf --enable-shared --disable-cxx
+        make
+        sudo make install
+        popd
+    - name: cmake build
+      run: |
+        set -x
+        gcc --version
+        echo 'export PATH=/home/runner/mpich/bin:$PATH' > .bashrc
+        source .bashrc
+        export CC=/home/runner/mpich/bin/mpicc
+        export FC=/home/runner/mpich/bin/mpifort
+        export LD_LIBRARY_PATH="/home/runner/netcdf-c/lib:/home/runner/mpich/lib:/home/runner/hdf5/lib:/home/runner/netcdf-fortran/lib:/home/runner/pnetcdf/lib:$LD_LIBRARY_PATH"
+        mkdir build
+        cd build
+        cmake -Wno-dev -DNetCDF_C_LIBRARY=/home/runner/netcdf-c/lib/libnetcdf.so -DNetCDF_C_INCLUDE_DIR=/home/runner/netcdf-c/include -DPnetCDF_PATH='/home/runner/pnetcdf' -DPIO_ENABLE_FORTRAN=Off -DPIO_ENABLE_LOGGING=On -DPIO_ENABLE_TIMING=Off .. || (cat CMakeFiles/CMakeOutput.log && cat CMakeFiles/CMakeError.log)
+        make VERBOSE=1
+        make tests VERBOSE=1
+#        ctest -VV
+    - name: cmake build
+      run: |
+        set -x
+        ls -l
+        echo 'export PATH=/home/runner/mpich/bin:$PATH' > .bashrc
+        source .bashrc
+        export CC=/home/runner/mpich/bin/mpicc
+        export FC=/home/runner/mpich/bin/mpifort
+        rm -rf b1
+        mkdir b1
+        cd b1
+        cmake -Wno-dev -DNetCDF_C_LIBRARY=/home/runner/netcdf/lib/libnetcdf.so -DNetCDF_C_INCLUDE_DIR=/usr/include -DPnetCDF_PATH='/home/runner/pnetcdf' -DPIO_ENABLE_FORTRAN=Off -DPIO_HDF5_LOGGING=On -DPIO_USE_MALLOC=On -DPIO_ENABLE_LOGGING=On -DPIO_ENABLE_TIMING=Off ..
+        make VERBOSE=1
+        make tests VERBOSE=1
+        ctest -VV
+        
+        
+        

--- a/configure.ac
+++ b/configure.ac
@@ -399,6 +399,14 @@ fi
 AM_CONDITIONAL(BUILD_NCINT, [test "x$enable_netcdf_integration" = xyes])
 AM_CONDITIONAL(NETCDF_INTEGRATION, [test "x$enable_netcdf_integration" = xyes])
 
+# If we are building netCDF integration and also then PIO Fortran
+# library, then we also need netcdf-fortran.
+if test "x$enable_netcdf_integration" = xyes -a "x$enable_fortran" = xyes; then
+   AC_LANG_PUSH([Fortran])
+   AC_CHECK_LIB([netcdff], [nf_inq_libvers], [], [AC_MSG_ERROR([Can't find or link to the netcdf-fortran library, required because both --enable-fortran and --enable-netcdf-integration are specified.])])
+   AC_LANG_POP([Fortran])
+fi
+
 AC_CONFIG_FILES([tests/general/pio_tutil.F90:tests/general/util/pio_tutil.F90])
 
 # The user may have changed the MPIEXEC for these test scripts.

--- a/tests/cunit/CMakeLists.txt
+++ b/tests/cunit/CMakeLists.txt
@@ -153,9 +153,9 @@ add_dependencies (tests test_async_perf)
 
 # Test Timeout in seconds.
 if (PIO_VALGRIND_CHECK)
-  set (DEFAULT_TEST_TIMEOUT 480)
+  set (DEFAULT_TEST_TIMEOUT 800)
 else ()
-  set (DEFAULT_TEST_TIMEOUT 480)
+  set (DEFAULT_TEST_TIMEOUT 600)
 endif ()
 
 # All tests need a certain number of tasks, but they should be able to

--- a/tests/cunit/test_darray_multivar.c
+++ b/tests/cunit/test_darray_multivar.c
@@ -580,15 +580,17 @@ int main(int argc, char **argv)
             if ((ret = PIOc_Init_Intracomm(test_comm, TARGET_NTASKS, ioproc_stride,
                                            ioproc_start, rearranger[r], &iosysid)))
                 return ret;
-
+            printf("test Rearranger %d\n",rearranger[r]);
             /* Run tests. */
             if ((ret = test_all_darray(iosysid, num_flavors, flavor, my_rank, test_comm,
                                        rearranger[r])))
                 return ret;
+            printf("test Rearranger %d complete\n",rearranger[r]);
 
             /* Finalize PIO system. */
             if ((ret = PIOc_free_iosystem(iosysid)))
                 return ret;
+
         }
 
     } /* endif my_rank < TARGET_NTASKS */

--- a/tests/cunit/test_darray_multivar.c
+++ b/tests/cunit/test_darray_multivar.c
@@ -580,12 +580,12 @@ int main(int argc, char **argv)
             if ((ret = PIOc_Init_Intracomm(test_comm, TARGET_NTASKS, ioproc_stride,
                                            ioproc_start, rearranger[r], &iosysid)))
                 return ret;
-            printf("test Rearranger %d\n",rearranger[r]);
+            /* printf("test Rearranger %d\n",rearranger[r]); */
             /* Run tests. */
             if ((ret = test_all_darray(iosysid, num_flavors, flavor, my_rank, test_comm,
                                        rearranger[r])))
                 return ret;
-            printf("test Rearranger %d complete\n",rearranger[r]);
+            /* printf("test Rearranger %d complete\n",rearranger[r]); */
 
             /* Finalize PIO system. */
             if ((ret = PIOc_free_iosystem(iosysid)))

--- a/tests/ncint/tst_var_compress.c
+++ b/tests/ncint/tst_var_compress.c
@@ -69,12 +69,13 @@ run_var_compress_test(int my_rank, int ntasks, int iosysid)
     if (nc_close(ncid)) PERR;
 
     {
-	int shuffle_in, deflate_in, deflate_level_in, storage_in;
+	/* int shuffle_in, deflate_in, deflate_level_in; */
+	int storage_in;
 	int *data_in;
 	size_t chunksizes_in[NDIM3];
 	int endian_in;
 	int d;
-        int ret;
+        /* int ret; */
 	
 	/* Open the file. */
 	if (nc_open(FILE_NAME, NC_PIO, &ncid)) PERR;

